vyper/semantics/types/function.py

```diff
@@ -752,7 +767,10 @@ def _parse_decorators(
             raise StructureException("Bad decorator syntax", decorator)
 
     if function_visibility is None:
-        function_visibility = FunctionVisibility.INTERNAL
+        if is_interface:
```

Comment:
i think we should actually bubble up `None` for function_visibility and let the caller handle it

Reply:
addressed in: 1ee63b6758b800af4ab63cce7176f725bfa9f7eb